stories/guides/CustomView.stories.mdx

```diff
@@ -0,0 +1,116 @@
+import { Meta } from '@storybook/addon-docs'
+import LinkTo from '@storybook/addon-links/react'
+
+<Meta title="Guides/Creating Custom Views" />
+
+# Creating A Custom View
+
+It is possible to override an existing Calendar `view`, or even create a new `view` all together, by adhering to a specific interface.
+
+<LinkTo kind="examples" story="example-9">
+  View An Example
+</LinkTo>
+
+First build your custom `view` component. This could be based off an existing component, or something completely custom.
+
+```js
+import React from 'react'
+import PropTypes from 'prop-types'
+import { Navigate as navigate } from 'react-big-calendar'
+import TimeGrid from 'react-big-calendar/lib/TimeGrid'
+
+export default function CustomWeekView({
+  date,
+  localizer,
+  max = localizer.endOf(new Date(), 'day'),
+  min = localizer.startOf(new Date(), 'day'),
+  scrollToTime = localizer.startOf(new Date(), 'day'),
+  ...props
+}) {
+  const currRange = useMemo(
+    () => CustomWeekView.range(date, { localizer }),
+    [date, localizer]
+  )
+
+  return (
+    <TimeGrid
+      date={date}
+      eventOffset={15}
+      localizer={localizer}
+      max={max}
+      min={min}
+      range={currRange}
+      scrollToTime={scrollToTime}
+      {...props}
+    />
+  )
+}
+
+CustomWeekView.propTypes = {
+  date: PropTypes.instanceOf(Date).isRequired,
+  localizer: PropTypes.object,
+  max: PropTypes.instanceOf(Date),
+  min: PropTypes.instanceOf(Date),
+  scrollToTime: PropTypes.instanceOf(Date),
+}
+
+CustomWeekView.range = (date, { localizer }) => {
+  const start = date
+  const end = dates.add(start, 2, 'day')
+
+  let current = start
+  const range = []
+
+  while (localizer.lte(current, end, 'day')) {
+    range.push(current)
+    current = localizer.add(current, 1, 'day')
+  }
+
+  return range
+}
+
+CustomWeekView.navigate = (date, action, { localizer }) => {
+  switch (action) {
+    case Navigate.PREVIOUS:
+      return localizer.add(date, -3, 'day')
+
+    case Navigate.NEXT:
+      return localizer.add(date, 3, 'day')
+
+    default:
+      return date
+  }
+}
+
+CustomWeekView.title = (date, { localizer }) => {
+  const [start, ...rest] = Week.range(date, { localizer })
```

Comment:
I think `Week` needs to be renamed to `CustomWeekView`. `Week` is not defined and I think we are trying to use the `range` method of our custom component?